src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -79,11 +79,11 @@ public function process(ContainerBuilder $container)
                     $container->getParameterBag()->mergeEnvPlaceholders($resolvingBag);
                 }
 
-                throw $e;
-            } finally {
```

Comment:
does this struct need to change? Before we'd reset placeholders in both cases (after try, after catch).

e.g. the state was self-cleaning independently, whereas now state is coupled a few lines below (L99) IIUC.

Reply:
We stack the available placeholder unique prefixes in BaseNode and only clean on exception and at the end of the process - if we clean after every extension process, we will never be able to stack.